### ISSUE TITLE
uglifyjs tries to copy/write a map file that does not exist and fails the production build

### ISSUE THREAD

**Issue by rolandjitsu** [driftyco/ionic-app-scripts#884]
**Note: for support questions, please use one of these channels:**

https://forum.ionicframework.com/
http://ionicworldwide.herokuapp.com/

#### Short description of the problem:

After I upgraded my app from Ionic `v2` to `v3` and app scripts to `v1.3.0`, the `ionic-app-scripts build --prod` seems to fail.

At first look, the error provides no information:
```text
> @infarm/maintenance@3.0.0 ionic:build /Users/rolandjitsu/Projects/infarm/maintenance
> ionic-app-scripts build

[21:11:15]  ionic-app-scripts 1.3.0 
[21:11:16]  build prod started ... 
[21:11:16]  clean started ... 
[21:11:16]  clean finished in 7 ms 
[21:11:16]  copy started ... 
[21:11:16]  ngc started ... 
[21:11:36]  ngc finished in 20.22 s 
[21:11:36]  preprocess started ... 
[21:11:36]  deeplinks started ... 
[21:11:36]  deeplinks finished in 692 ms 
[21:11:36]  optimization started ... 
[21:11:37]  copy finished in 21.23 s 
[21:11:55]  optimization finished in 18.77 s 
[21:11:55]  preprocess finished in 19.46 s 
[21:11:55]  rollup started ... 
[21:12:08]  rollup: 
            commonjs-proxy:/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@angular/core/@angular/core.es5.js
            has issued a warning: 'default' is not exported by 'node_modules/@angular/core/@angular/core.es5.js' 
[21:12:08]  rollup: 
            commonjs-proxy:/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@angular/platform-browser/@angular/platform-browser/animations.es5.js
            has issued a warning: 'default' is not exported by 
            'node_modules/@angular/platform-browser/@angular/platform-browser/animations.es5.js' 
[21:12:08]  rollup: 
            commonjs-proxy:/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@angular/animations/@angular/animations.es5.js
            has issued a warning: 'default' is not exported by 
            'node_modules/@angular/animations/@angular/animations.es5.js' 
[21:12:08]  rollup: 
            commonjs-proxy:/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@angular/http/@angular/http.es5.js
            has issued a warning: 'default' is not exported by 'node_modules/@angular/http/@angular/http.es5.js' 
[21:12:08]  rollup: 
            commonjs-proxy:/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@angular/common/@angular/common.es5.js
            has issued a warning: 'default' is not exported by 'node_modules/@angular/common/@angular/common.es5.js' 
[21:12:13]  rollup finished in 18.04 s 
[21:12:13]  sass started ... 
[21:12:13]  uglifyjs started ... 
[21:12:15]  sass finished in 1.47 s 
[21:12:15]  cleancss started ... 
[21:12:17]  cleancss finished in 1.78 s 
[21:12:38]  ionic-app-script task: "build" 
[21:12:38]  Error: [object Object] 
Error: [object Object]
    at new BuildError (/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@ionic/app-scripts/dist/util/errors.js:16:28)
    at ChildProcess.<anonymous> (/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@ionic/app-scripts/dist/worker-client.js:38:24)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:752:12)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

But I then added a `console.log()` in `utils/error.js` at `L16` and saw the following:
```text
> @infarm/maintenance@3.0.0 ionic:build /Users/rolandjitsu/Projects/infarm/maintenance
> ionic-app-scripts build "--device" "--prod"

[21:14:44]  ionic-app-scripts 1.3.0 
[21:14:44]  build prod started ... 
[21:14:44]  clean started ... 
[21:14:44]  clean finished in 3 ms 
[21:14:44]  copy started ... 
[21:14:44]  ngc started ... 
[21:15:07]  ngc finished in 22.95 s 
[21:15:07]  preprocess started ... 
[21:15:07]  deeplinks started ... 
[21:15:08]  deeplinks finished in 1.27 s 
[21:15:08]  optimization started ... 
[21:15:09]  copy finished in 24.84 s 
[21:15:28]  optimization finished in 20.09 s 
[21:15:28]  preprocess finished in 21.36 s 
[21:15:28]  rollup started ... 
[21:15:43]  rollup: 
            commonjs-proxy:/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@angular/core/@angular/core.es5.js
            has issued a warning: 'default' is not exported by 'node_modules/@angular/core/@angular/core.es5.js' 
[21:15:43]  rollup: 
            commonjs-proxy:/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@angular/platform-browser/@angular/platform-browser/animations.es5.js
            has issued a warning: 'default' is not exported by 
            'node_modules/@angular/platform-browser/@angular/platform-browser/animations.es5.js' 
[21:15:43]  rollup: 
            commonjs-proxy:/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@angular/animations/@angular/animations.es5.js
            has issued a warning: 'default' is not exported by 
            'node_modules/@angular/animations/@angular/animations.es5.js' 
[21:15:43]  rollup: 
            commonjs-proxy:/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@angular/http/@angular/http.es5.js
            has issued a warning: 'default' is not exported by 'node_modules/@angular/http/@angular/http.es5.js' 
[21:15:43]  rollup: 
            commonjs-proxy:/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@angular/common/@angular/common.es5.js
            has issued a warning: 'default' is not exported by 'node_modules/@angular/common/@angular/common.es5.js' 
[21:15:49]  rollup finished in 20.34 s 
[21:15:49]  sass started ... 
[21:15:49]  uglifyjs started ... 
[21:15:50]  sass finished in 1.55 s 
[21:15:50]  cleancss started ... 
[21:15:52]  cleancss finished in 1.66 s 
ENOENT: no such file or directory, open '/Users/rolandjitsu/Projects/infarm/maintenance/www/build/shim.js.map'
{ message: 'ENOENT: no such file or directory, open \'/Users/rolandjitsu/Projects/infarm/maintenance/www/build/shim.js.map\'',
  name: 'Error',
  stack: 'Error: ENOENT: no such file or directory, open \'/Users/rolandjitsu/Projects/infarm/maintenance/www/build/shim.js.map\'\n    at new BuildError (/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@ionic/app-scripts/dist/util/errors.js:17:28)\n    at taskReject (/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@ionic/app-scripts/dist/worker-process.js:36:22)\n    at /Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@ionic/app-scripts/dist/worker-process.js:14:13\n    at process._tickCallback (internal/process/next_tick.js:103:7)',
  hasBeenLogged: false,
  isFatal: false }
[21:16:15]  ionic-app-script task: "build" 
[21:16:15]  Error: [object Object] 
{ Error: [object Object]
    at new BuildError (/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@ionic/app-scripts/dist/util/errors.js:17:28)
    at ChildProcess.<anonymous> (/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@ionic/app-scripts/dist/worker-client.js:38:24)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:752:12)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9) hasBeenLogged: false, isFatal: false }
Error: [object Object]
    at new BuildError (/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@ionic/app-scripts/dist/util/errors.js:17:28)
    at ChildProcess.<anonymous> (/Users/rolandjitsu/Projects/infarm/maintenance/node_modules/@ionic/app-scripts/dist/worker-client.js:38:24)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:752:12)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

Which tells me that some process is trying to copy `www/build/shim.js.map` or maybe trying to write it and it cannot.

I am actually copying some files from [core-js](https://github.com/zloirock/core-js) which are needed to provided fallbacks for browsers (devices) that don't support all ES6 features:
```js
// This is a custom dictionary to make it easy to extend/override.
// Provide a name for an entry, it can be anything such as 'copyAssets' or 'copyFonts',
// then provide an object with a `src` array of globs and a `dest` string.
// Source: https://github.com/driftyco/ionic-app-scripts/blob/master/config/copy.config.js
module.exports = {
	copyIndexContent: {
		src: ['{{SRC}}/index.html', '{{SRC}}/manifest.json', '{{SRC}}/service-worker.js'],
		dest: '{{WWW}}'
	},
	// Some packages and source code may use ES6 features,
	// therefore we provide shims/polyfills for ES6 features via core-js.
	// TODO: Eventually we will remove core-js once ES6 is stable enough
	copyPolyfills: {
		src: [
			'{{ROOT}}/node_modules/core-js/client/shim.js',
			'{{ROOT}}/node_modules/core-js/client/shim.min.js',
			'{{ROOT}}/node_modules/core-js/client/shim.min.js.map',
			'{{ROOT}}/node_modules/ionic-angular/polyfills/polyfills.js'
		],
		dest: '{{BUILD}}'
	},
	copySwToolbox: {
		src: ['{{ROOT}}/node_modules/sw-toolbox/sw-toolbox.js'],
		dest: '{{BUILD}}'
	},
	copyFonts: {
		src: ['{{ROOT}}/node_modules/ionicons/dist/fonts/**/*', '{{ROOT}}/node_modules/ionic-angular/fonts/**/*'],
		dest: '{{WWW}}/assets/fonts'
	},
	copyAssets: {
		src: ['{{SRC}}/assets/**/*'],
		dest: '{{WWW}}/assets'
	},
	// NOTE: Firebase plugin needs these files.
	copyFirebaseConfigFiles: {
		src: [
			'{{ROOT}}/google-services.json',
			'{{ROOT}}/GoogleService-Info.plist'
		],
		dest: '{{WWW}}'
	}
};
```

#### What behavior are you expecting?

It should build without issues. It should also provide a more descriptive error message than what I see without adding the log myself.

**Steps to reproduce:**
1.
2.
3.

```
insert any relevant code between the above and below backticks
```

**Which @ionic/app-scripts version are you using?**

I'm using `v1.3.0`. See my `package.json`:
```json
{
	"name": "",
	"version": "0.0.0",
	"description": "",
	"keywords": [],
	"homepage": "",
	"author": {
		"name": "Roland Groza",
		"email": "rolandjitsu@gmail.com"
	},
	"scripts": {
		"ionic:build": "ionic-app-scripts build",
		"ionic:serve": "ionic-app-scripts serve",
		"ionic:clean": "ionic-app-scripts clean",
		"ionic:lint": "ionic-app-scripts lint"
	},
	"config": {
		"ionic_bundler": "rollup",
		"ionic_rollup": "./config/rollup.config.js",
		"ionic_copy": "./config/copy.config.js",
		"ionic_experimental_parse_deeplinks": true,
		"ionic_experimental_purge_decorators": true
	},
	"dependencies": {
		"@angular/animations": "4.0.0",
		"@angular/common": "4.0.0",
		"@angular/compiler": "4.0.0",
		"@angular/compiler-cli": "4.0.0",
		"@angular/core": "4.0.0",
		"@angular/forms": "4.0.0",
		"@angular/http": "4.0.0",
		"@angular/platform-browser": "4.0.0",
		"@angular/platform-browser-dynamic": "4.0.0",
		"@ionic-native/background-geolocation": "3.4.2",
		"@ionic-native/camera": "3.4.2",
		"@ionic-native/core": "3.4.2",
		"@ionic-native/firebase": "3.4.2",
		"@ionic-native/geolocation": "3.4.2",
		"@ionic-native/keyboard": "3.4.2",
		"@ionic-native/nfc": "3.4.2",
		"@ionic-native/splash-screen": "3.4.2",
		"@ionic-native/status-bar": "3.4.2",
		"@ionic/storage": "2.0.1",
		"core-js": "^2.4.1",
		"ionic-angular": "3.0.1",
		"ionicons": "3.0.0",
		"moment": "^2.18.1",
		"raven-js": "^3.14.0",
		"reflect-metadata": "^0.1.9",
		"rxjs": "5.1.1",
		"sw-toolbox": "3.6.0",
		"zone.js": "^0.8.5"
	},
	"devDependencies": {
		"@angular/compiler-cli": "4.0.0",
		"@angular/tsc-wrapped": "4.0.0",
		"@ionic/app-scripts": "1.3.0",
		"@types/node": "^7.0.12",
		"chalk": "^1.1.3",
		"codelyzer": "3.0.0-beta.4",
		"conventional-changelog": "^1.1.3",
		"cordova": "^6.5.0",
		"fs-extra": "^2.1.2",
		"googleapis": "^18.0.0",
		"ionic": "^2.2.2",
		"ionic-app-lib": "^2.2.1",
		"ip": "^1.1.5",
		"nodegit": "^0.18.0",
		"rollup-plugin-replace": "^1.1.1",
		"semver": "^5.3.0",
		"tslint": "^4.5.1",
		"typescript": "^2.2.2",
		"uuid": "^3.0.1",
		"xml2js": "^0.4.17",
		"yargs": "^7.0.2"
	},
	"engines": {
		"node": ">= 7.4"
	}
}
```

**Other information:** (e.g. stacktraces, related issues, suggestions how to fix, stackoverflow links, forum links, etc)

**Comment by rolandjitsu** [comment]
Enabling `"ionic_use_experimental_closure": true` enforces my suspicion of UglifyJS failing as switching to Closure seems to work.

**Comment by danbucholtz** [comment]
Fixed in `1.3.2`.

Thanks,
Dan
